server/services/allocations.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable camelcase */
+// disabling camelcase check so that we can manipulate snake_case attributes without changing structure
+const { dbClient, collections } = require('../db');
+
+/**
+ *
```

Comment:
I like that there's a type being defined, but I don't see any references being made to it yet. Is that going to be done in another commit? I'm also wondering if there's a clearer, more unique type name than `allocation` to give it, but that's a nitpick.

Reply:
I think perhaps I am confused, I was using this as a doc block to define what is being returned from `getPhaseAllocation`, What am I missing?

Reply:
`@typedef` is a JSDoc tag which *defines* a type. It does not mark any given symbols as that type. For that, you need `@type`. Your JSDoc block is equivalent to the following TypeScript:

```typescript
type allocation {
  id: number;
  phase_id: string;
  allocation: number;
  site_id: number;
}
```

In other words, you're making a new *type* at the file scope, which happens to share a name with a local *variable*, but has nothing to do with it. We can check this by hovering over the `allocation` variable:
![image](https://user-images.githubusercontent.com/123127934/219469574-8c7d98f6-cc03-4e28-b562-d980dc2f244b.png)

This can be fixed by tagging the variable with `@type {typename}`, like so:
![image](https://user-images.githubusercontent.com/123127934/219470138-7b536cd9-8562-4062-a4b8-d3237c580b74.png)

And now, VSCode and other IDEs can provide information as you edit, and TS can check typing properly:
![image](https://user-images.githubusercontent.com/123127934/219470530-ae3caa2f-d559-4cd7-8ab8-906c1dfae265.png)
![image](https://user-images.githubusercontent.com/123127934/219470667-8716bfd4-c741-46ce-8df3-c6b58a8aa9fe.png)

Reply:
ah - okay thank you for the detailed explanation, I think I have it working the correct way now.